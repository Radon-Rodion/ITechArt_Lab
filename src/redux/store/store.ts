import { combineReducers, createStore } from "redux";
import userReducer from "./reducers/userReducer";

const rootReducer = combineReducers({
  user: userReducer,
});

const store = createStore(rootReducer);

export type RootState = ReturnType<typeof rootReducer>;

export default store;
