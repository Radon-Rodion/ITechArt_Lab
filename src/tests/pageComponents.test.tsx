/**
 * @jest-environment jsdom
 */

import React from "react";
import { render } from "@testing-library/react";
import { Provider } from "react-redux";
import { BrowserRouter } from "react-router-dom";
import store from "@/redux/store/store";
import Header from "@/components/header/header";
import Footer from "@/components/footer/footer";

describe("Header and footer snapshots", () => {
  test("Header", () => {
    const header = render(
      <Provider store={store}>
        <BrowserRouter>
          <Header />
        </BrowserRouter>
      </Provider>
    );
    expect(header).toMatchSnapshot("HeaderSnapshoot");
  });

  test("Footer", () => {
    const footer = render(<Footer />);
    expect(footer).toMatchSnapshot("FooterSnapshoot");
  });
});
