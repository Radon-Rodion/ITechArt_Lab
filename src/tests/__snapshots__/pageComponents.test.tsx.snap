// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header and footer snapshots Footer: FooterSnapshoot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <footer>
        <div
          class="slogan"
        >
          "Games are not just hobbies for us!"
        </div>
        <ul
          class="list"
        >
          <li
            class="element"
          >
            <a
              class="footerNavButton"
              href="https://vironit.com/"
            >
              <img
                alt="VironIT"
                class="footerNavButtonImage"
                src=""
              />
            </a>
          </li>
          <li
            class="element"
          >
            <a
              class="footerNavButton"
              href="https://playgendary.com/ru"
            >
              <img
                alt="PlayGendary"
                class="footerNavButtonImage"
                src=""
              />
            </a>
          </li>
          <li
            class="element"
          >
            <a
              class="footerNavButton"
              href="http://www.herocraft.com/"
            >
              <img
                alt="HeroCraft"
                class="footerNavButtonImage"
                src=""
              />
            </a>
          </li>
        </ul>
      </footer>
    </div>
  </body>,
  "container": <div>
    <footer>
      <div
        class="slogan"
      >
        "Games are not just hobbies for us!"
      </div>
      <ul
        class="list"
      >
        <li
          class="element"
        >
          <a
            class="footerNavButton"
            href="https://vironit.com/"
          >
            <img
              alt="VironIT"
              class="footerNavButtonImage"
              src=""
            />
          </a>
        </li>
        <li
          class="element"
        >
          <a
            class="footerNavButton"
            href="https://playgendary.com/ru"
          >
            <img
              alt="PlayGendary"
              class="footerNavButtonImage"
              src=""
            />
          </a>
        </li>
        <li
          class="element"
        >
          <a
            class="footerNavButton"
            href="http://www.herocraft.com/"
          >
            <img
              alt="HeroCraft"
              class="footerNavButtonImage"
              src=""
            />
          </a>
        </li>
      </ul>
    </footer>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Header and footer snapshots Header: HeaderSnapshoot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <header
        class="header"
      >
        <div
          class="title"
        >
          Game Store
        </div>
        <ul
          class="navbar"
        >
          <nav>
            <li
              class="navbutton"
            >
              <a
                aria-current="page"
                class="active"
                href="/"
              >
                Home
              </a>
            </li>
            <li
              class="navbutton"
            >
              <button
                class="notButton navlink listHeader"
                type="button"
              >
                Products
              </button>
              <ul
                class="list"
              >
                <li
                  class="listItem"
                >
                  <button
                    class="notButton link"
                    type="button"
                  >
                    PC
                  </button>
                </li>
                <li
                  class="listItem"
                >
                  <button
                    class="notButton link"
                    type="button"
                  >
                    Playstation
                  </button>
                </li>
                <li
                  class="listItem"
                >
                  <button
                    class="notButton link"
                    type="button"
                  >
                    Xbox
                  </button>
                </li>
              </ul>
            </li>
            <li
              class="navbutton"
            >
              <button
                class="notButton navlink"
                type="button"
              >
                About
              </button>
            </li>
            <li
              class="navbutton"
            >
              <button
                class="notButton navbarBtn"
                type="button"
              >
                Sign In
              </button>
            </li>
            <li
              class="navbutton"
            >
              <button
                class="notButton navbarBtn"
                type="button"
              >
                Sign Up
              </button>
            </li>
          </nav>
        </ul>
      </header>
    </div>
  </body>,
  "container": <div>
    <header
      class="header"
    >
      <div
        class="title"
      >
        Game Store
      </div>
      <ul
        class="navbar"
      >
        <nav>
          <li
            class="navbutton"
          >
            <a
              aria-current="page"
              class="active"
              href="/"
            >
              Home
            </a>
          </li>
          <li
            class="navbutton"
          >
            <button
              class="notButton navlink listHeader"
              type="button"
            >
              Products
            </button>
            <ul
              class="list"
            >
              <li
                class="listItem"
              >
                <button
                  class="notButton link"
                  type="button"
                >
                  PC
                </button>
              </li>
              <li
                class="listItem"
              >
                <button
                  class="notButton link"
                  type="button"
                >
                  Playstation
                </button>
              </li>
              <li
                class="listItem"
              >
                <button
                  class="notButton link"
                  type="button"
                >
                  Xbox
                </button>
              </li>
            </ul>
          </li>
          <li
            class="navbutton"
          >
            <button
              class="notButton navlink"
              type="button"
            >
              About
            </button>
          </li>
          <li
            class="navbutton"
          >
            <button
              class="notButton navbarBtn"
              type="button"
            >
              Sign In
            </button>
          </li>
          <li
            class="navbutton"
          >
            <button
              class="notButton navbarBtn"
              type="button"
            >
              Sign Up
            </button>
          </li>
        </nav>
      </ul>
    </header>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
